mcproxy/patches/0006-block-ingress.patch

```diff
@@ -0,0 +1,93 @@
+Index: mcproxy-2017-08-24-93b5ace42268160ebbfff4c61818fb15fa2d9b99/mcproxy/src/proxy/proxy_instance.cpp
+===================================================================
+--- mcproxy-2017-08-24-93b5ace42268160ebbfff4c61818fb15fa2d9b99.orig/mcproxy/src/proxy/proxy_instance.cpp
++++ mcproxy-2017-08-24-93b5ace42268160ebbfff4c61818fb15fa2d9b99/mcproxy/src/proxy/proxy_instance.cpp
+@@ -171,6 +171,9 @@ void proxy_instance::worker_thread()
+     HC_LOG_TRACE("");
+     while (m_running) {
+         auto msg = m_job_queue.dequeue();
++
++        HC_LOG_DEBUG("Proxy Message: " << msg->get_message_type_name(msg->get_type()) );
++
+         switch (msg->get_type()) {
+         case proxy_msg::TEST_MSG:
+             (*msg)();
+@@ -193,25 +196,69 @@ void proxy_instance::worker_thread()
+         }
+         break;
+         case proxy_msg::GROUP_RECORD_MSG: {
+-            auto r =  std::static_pointer_cast<group_record_msg>(msg);
++            auto gr =  std::static_pointer_cast<group_record_msg>(msg);
+ 
+             if (m_in_debug_testing_mode) {
+                 std::cout << "!!--ACTION: receive record" << std::endl;
+-                std::cout << *r << std::endl;
++                std::cout << *gr << std::endl;
+                 std::cout << std::endl;
+             }
+ 
+-            auto it = m_downstreams.find(r->get_if_index());
++            auto slist = gr->get_slist();
++            addr_storage saddr;
++            if ( slist.empty() )
++            {
```

Comment:
The code style changes here. Better keep in uniform.

Reply:
the mcproxy code is littered with 25 different coding styles, but if it makes you happy :-)

Reply:
ok, it is one of those projects. :P